RELEASE.md

```diff
@@ -0,0 +1,45 @@
+# Releasing connect-go
+
+This document outlines how to create a release of connect-go.
+
+1. Clone the repo, ensuring you have the latest main.
+
+2. On a new branch, open [connect.go](connect.go) and change the `Version` constant to an appropriate [semantic version](https://semver.org/). To select the correct version, look at the version number of the [latest release](https://github.com/connectrpc/connect-go/releases/latest) and the changes that are included in this new release.
+  * If there are only bug fixes and no new features, remove the `-dev` suffix, set MINOR number to be equal to the [latest release](https://github.com/connectrpc/connect-go/releases/latest), and set the PATCH number to be 1 more than the PATCH number of the [latest release](https://github.com/connectrpc/connect-go/releases/latest).
+  * If there are features being released, remove the `-dev` suffix, set the MINOR number to be 1 more than the MINOR number of the [latest release](https://github.com/connectrpc/connect-go/releases/latest), and set the PATCH number to `0`. In the common case, the diff here will just be to remove the `-dev` suffix.
+
+   ```patch
+   -const Version = "1.14.0-dev"
+   +const Version = "1.14.0"
+   ```
+
+3. Check [cmd/protoc-gen-connect-go/main.go](cmd/protoc-gen-connect-go/main.go) to see if a new `IsAtLeastVersionX_Y_Z` constant was added since the [latest release](https://github.com/connectrpc/connect-go/releases/latest). If one was added (this is rare), ensure that the version number matches the version number that you are currently releasing. If it doesn't match, update the name to match the version you are about to release. [Example PR #496](https://github.com/connectrpc/connect-go/pull/496).
+
+4. Open a PR titled "Prepare for vX.Y.Z" ([Example PR #661](https://github.com/connectrpc/connect-go/pull/661)). Once it's reviewed and CI passes, merge it.
+
+    *Make sure no new commits are merged until the release is complete.*
```

Comment:
This one's a bit tricky. I think we may want to actually tag _all_ maintainers in the "prepare" PR, to tell them to not approve/merge any commits until the "back to dev" PR lands.

Reply:
I've updated 4. to say that the description of the PR should tag all current maintainers.